server/channels/store/sqlstore/post_store.go

```diff
@@ -1503,6 +1503,116 @@ func (s *SqlPostStore) GetPostsAfter(rctx request.CTX, options model.GetPostsOpt
 	return s.getPostsAround(rctx, false, options, sanitizeOptions)
 }
 
+func (s *SqlPostStore) GetPostsForReporting(rctx request.CTX, queryParams model.ReportPostQueryParams) (*model.ReportPostListResponse, error) {
+	// Validate required parameters
+	if queryParams.ChannelId == "" {
+		return nil, store.NewErrInvalidInput("Post", "ChannelId", "")
+	}
+
+	// Convert model field names to SQL column names
+	timeField := "CreateAt"
+	if queryParams.TimeField == model.ReportingTimeFieldUpdateAt {
+		timeField = "UpdateAt"
+	}
+
+	sortDirection := "ASC"
+	if queryParams.SortDirection == model.ReportingSortDirectionDesc {
+		sortDirection = "DESC"
+	}
+
+	// Build base query - request one extra to determine if there are more pages
+	query := s.getQueryBuilder().
+		Select("*").
```

Comment:
Something I missed before: we are trying to avoid selecting everything unconditionally in queries. See https://mattermost.atlassian.net/browse/MM-62110.